Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerAttachmentsView.swift

```diff
@@ -5,91 +5,163 @@
+    open var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
+    open var documentPreviewItemHeight: CGFloat = 70
+    open var maxNumberOfVisibleDocuments: Int = 3
+    
+    override open var intrinsicContentSize: CGSize {
+        switch state {
+        case .media:
+            let height = imagePreviewItemSize.height + layoutMargins.top + layoutMargins.bottom
+            return .init(
+                width: UIView.noIntrinsicMetric,
+                height: height
+            )
+        case let .documents(items):
+            var height: CGFloat
+            if items.count > maxNumberOfVisibleDocuments {
+                let itemsHeight = documentPreviewItemHeight * CGFloat(maxNumberOfVisibleDocuments)
+                let spacings = minimumInteritemSpacing * CGFloat(maxNumberOfVisibleDocuments)
+                height = itemsHeight + spacings + layoutMargins.top + layoutMargins.bottom
+            } else {
+                height = collectionView.contentSize.height
+            }
+            return .init(
+                width: UIView.noIntrinsicMetric,
+                height: height
+            )
+        }
+    }
+    
+    public var state: State = .media(previews: []) {
         didSet {
-            collectionView.reloadData()
+            updateContentIfNeeded()
         }
     }
     
     var didTapRemoveItemButton: ((Int) -> Void)?
     
-    // MARK: - Subviews
-    
-    public private(set) lazy var flowLayout: UICollectionViewFlowLayout = {
-        let flowLayout = UICollectionViewFlowLayout()
-        flowLayout.itemSize = .init(width: 50, height: 70)
-        flowLayout.sectionInset = .init(top: 20, left: 10, bottom: 10, right: 10)
-        flowLayout.minimumInteritemSpacing = 10
-        flowLayout.scrollDirection = .horizontal
+    open var flowLayout: UICollectionViewFlowLayout {
+        var flowLayout = UICollectionViewFlowLayout()
         
-        return flowLayout
-    }()
-    
-    public private(set) lazy var collectionView: UICollectionView = .init(frame: .zero, collectionViewLayout: flowLayout)
-
-    // MARK: - Init
-    
-    public required init(
-        uiConfig: UIConfig<ExtraData> = .default
-    ) {
-        self.uiConfig = uiConfig
+        switch state {
+        case .documents:
+            let documentLayout = uiConfig
+                .messageComposer
+                .documentAttachmentsFlowLayout
+                .init()
+            
+            documentLayout.itemHeight = documentPreviewItemHeight
+            flowLayout = documentLayout
+        case .media:
+            flowLayout = UICollectionViewFlowLayout()
+            flowLayout.itemSize = imagePreviewItemSize
+            flowLayout.scrollDirection = .horizontal
+        }
         
-        super.init(frame: .zero)
+        flowLayout.sectionInset = layoutMargins
+        flowLayout.minimumInteritemSpacing = minimumInteritemSpacing
         
-        commonInit()
+        return flowLayout
     }
```

Comment:
This is a computed variable so you create a new instance of layout every time you touch it.